app/models/name/observations.rb

```diff
@@ -38,12 +38,12 @@ def of_taxon_any_name
 
     # "Observations of other taxa, this taxon proposed"
     def where_taxon_proposed
-      @all.reject { |obs| obs&.name_id == @name.id }
+      @all.reject { |obs| obs&.name_id.in?(@name_ids) }
     end
 
     # "Observations of any taxon, this name proposed"
     def where_name_proposed
-      @all
+      @all.where(namings: { name_id: @name.id })
```

Comment:
@nimmolo We need the info from `namings`, not just from the obss.
I hope this will have little or no effect on performance. I think the `namings` have been eager loaded. Is that right?

Reply:
That's right - check the definition of `all` above. Thanks for figuring this out!